From: Liu Jing <liujing@cmss.chinamobile.com>
Subject: tools/mm: Free the allocated memory
Date: Tue, 22 Oct 2024 09:25:26 +0800

The comm_str memory needs to be freed if the search_pattern function call
fails in get_comm

Link: https://lkml.kernel.org/r/20241022012526.7597-1-liujing@cmss.chinamobile.com
Signed-off-by: Liu Jing <liujing@cmss.chinamobile.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/mm/page_owner_sort.c |    1 +
 1 file changed, 1 insertion(+)

--- a/tools/mm/page_owner_sort.c~tools-mm-free-the-allocated-memory
+++ a/tools/mm/page_owner_sort.c
@@ -377,6 +377,7 @@ static char *get_comm(char *buf)
 	if (errno != 0) {
 		if (debug_on)
 			fprintf(stderr, "wrong comm in follow buf:\n%s\n", buf);
+		free (comm_str);
 		return NULL;
 	}
 
_
